src/NuGet.Core/NuGet.Protocol/Utility/MetadataReferenceCache.cs

```diff
@@ -17,6 +17,8 @@ public class MetadataReferenceCache
         private readonly Dictionary<string, string> _stringCache = new Dictionary<string, string>(StringComparer.Ordinal);
         private readonly Dictionary<Type, PropertyInfo[]> _propertyCache = new Dictionary<Type, PropertyInfo[]>();
         private readonly Dictionary<string, NuGetVersion> _versionCache = new Dictionary<string, NuGetVersion>(StringComparer.Ordinal);
+        private readonly Dictionary<Type, MethodInfo> _stringMethodsCache = new Dictionary<Type, MethodInfo>();
```

Comment:
I'm not sure what this is supposed to do. 

Can you add tests for the MetadataREferenceCache additions?

Reply:
It helps with reflection. We're doing reflection on with every run is expensive.

Reply:
Added unit test.